source/utilities.cc

```diff
@@ -1822,6 +1822,30 @@ namespace aspect
       return lookup->get_data(position,component);
     }
 
+
+    template<int dim>
+    double compute_spd_factor(const double eta,
+                              const SymmetricTensor<2,dim> &strain_rate,
+                              const SymmetricTensor<2,dim> &dviscosities_dstrain_rate)
+    {
+      double alpha = 0;
+      const double norm_a_b = std::sqrt((strain_rate*strain_rate)*(dviscosities_dstrain_rate*dviscosities_dstrain_rate));
+      const double contract_a_b = (strain_rate*dviscosities_dstrain_rate);
+      const double one_minus_part = 1 - (contract_a_b / norm_a_b);
+      const double denom = one_minus_part * one_minus_part * norm_a_b;
+      if (denom == 0)
+        alpha = 1.0;
+      else
+        {
+          alpha = (2.0*eta)/denom;
+          if (alpha >= 1.0)
+            alpha = 1.0;
+          else
+            alpha = std::max(0.0,0.9*alpha);
```

Comment:
Oh, actually you document neither the `max` operation, nor the factor of 0.9. Can you add that as well?